### PR TITLE
CIS Profiles for SLE12

### PR DESCRIPTION
**Pull request by truzzon**
#### Description:

- Initial CIS Profiles for SLE12. Based on Benchmark Version 3.0.0 (released at 2021-04-27)
- Minor updates to SLE15 Chapter reference, If I found them during work.

### CODE REVIEW

File: linux_os/guide/system/permissions/mounting/kernel_module_udf_disabled/rule.yml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-prodtype: fedora,rhcos4,rhel7,rhel8,rhel9,sle15,ubuntu1804,ubuntu2004
+prodtype: fedora,rhcos4,rhel7,rhel8,rhel9,sle15,ubuntu1804,ubuntu2004,sle12
```
Comment (Mab879) [inline linux_os/guide/system/permissions/mounting/kernel_module_udf_disabled/rule.yml:3]: Please alphabetize this list.

File: linux_os/guide/system/bootloader-grub2/non-uefi/file_permissions_grub2_cfg/rule.yml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-prodtype: fedora,ol7,ol8,rhel7,rhel8,rhel9,rhv4,sle15,ubuntu2004
+prodtype: fedora,ol7,ol8,rhel7,rhel8,rhel9,rhv4,sle15,ubuntu2004,sle12
```
Comment (Mab879) [inline linux_os/guide/system/bootloader-grub2/non-uefi/file_permissions_grub2_cfg/rule.yml:3]: Please alphabetize this list.

File: linux_os/guide/system/bootloader-grub2/non-uefi/file_owner_grub2_cfg/rule.yml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-prodtype: fedora,ol7,ol8,rhel7,rhel8,rhel9,rhv4,sle15,ubuntu2004
+prodtype: fedora,ol7,ol8,rhel7,rhel8,rhel9,rhv4,sle15,ubuntu2004,sle12
```
Comment (Mab879) [inline linux_os/guide/system/bootloader-grub2/non-uefi/file_owner_grub2_cfg/rule.yml:3]: Please alphabetize this list.

File: linux_os/guide/system/bootloader-grub2/non-uefi/file_groupowner_grub2_cfg/rule.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-prodtype: fedora,ol7,ol8,rhel7,rhel8,rhel9,rhv4,sle15
+prodtype: fedora,ol7,ol8,rhel7,rhel8,rhel9,rhv4,sle15,sle12
```
Comment (Mab879) [inline linux_os/guide/system/bootloader-grub2/non-uefi/file_groupowner_grub2_cfg/rule.yml:3]: Please alphabetize this list.

File: linux_os/guide/services/mail/service_postfix_enabled/rule.yml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-prodtype: ol7,ol8,rhel7,rhel8,rhel9,sle15
+prodtype: ol7,ol8,rhel7,rhel8,rhel9,sle12,sle15
```
Comment (Mab879) [inline linux_os/guide/services/mail/service_postfix_enabled/rule.yml:3]: Not seeing any references is that by design?

**Review by Mab879 — CHANGES_REQUESTED** [review]
Thank you for this contribution. I left few notes on this PR, mostly around `prodtype` ordering. Also please make sure that you have `cce` on the rules as well.

**Comment by truzzon** [comment]
Hello @Mab879, thanks for your feedback. I'll fix, what I can in the next days. 

About the CCE reference for SLE: there are none in the Nist resources. 
@brett060102 and @guangyee since your profile says SUSE, can you provide a CCE resource, where I can look up the missing references, if publicly available? I also have an SCC account if required.

File: linux_os/guide/services/mail/service_postfix_enabled/rule.yml (hunk 5 above)
Comment (truzzon) [reply linux_os/guide/services/mail/service_postfix_enabled/rule.yml:3]: To me, it seemed like a dependency.

`2.2.19 Ensure mail transfer agent is configured for local-only mode (Automated)` uses checks related to it. The build failed locally if I didn't add the prodtype.

**Comment by truzzon** [comment]
Hello @Mab879,

I have contacted @teacup-on-rockingchair via e-mail and asked him, if he could help me out with the CCEs, since he is the only one in the recent commit history, who added CCEs for SLE.
He responded, that he can't disclose the source. I'm currently waiting for any information, he can provide, or maybe a PR, that adds the CCEs to my changes.

In case, the references are incomplete or nothing can be provided, is it still possible to merge the PR? After all, the build is working. And on my past PR CCEs were not required at all for SLE.

**Comment by Mab879** [comment]
@truzzon Thanks for the update. Please rebase this PR and I will take a look again and get it merged.

**Review by Mab879 — APPROVED** [review]

File: linux_os/guide/services/ntp/ntpd_configure_restrictions/rule.yml [hunk 6]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-prodtype: fedora,rhel7
+prodtype: rhel7,fedora,sle12
```
Comment (Mab879) [inline linux_os/guide/services/ntp/ntpd_configure_restrictions/rule.yml:3]: Please alphabetize this list.

File: linux_os/guide/services/ntp/ntpd_run_as_ntp_user/rule.yml [hunk 7]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-prodtype: fedora,rhel7
+prodtype: rhel7,fedora,sle12
```
Comment (Mab879) [inline linux_os/guide/services/ntp/ntpd_run_as_ntp_user/rule.yml:3]: Please alphabetize this list.

**Review by Mab879 — CHANGES_REQUESTED** [review]
Please run `./utils/fix_rules.py sort_prodtypes` to fix the prodtype ordering issues.
See https://complianceascode.readthedocs.io/en/latest/manual/developer/05_tools_and_utilities.html for more information on running the utilities.

**Comment by truzzon** [comment]
Hello @Mab879,

I fixed your findings manually. The scripts did not work for me yet:
```
(content_3.9.6) nico@tower-nico: content [cis_sle12_v300]$ ./utils/rule_dir_json.py 
Product sle12 has multiple remediations of the same type in rule no_empty_passwords: shared.sh,shared.yml,sle12.yml,sle15.yml
Product sle15 has multiple remediations of the same type in rule no_empty_passwords: shared.sh,shared.yml,sle12.yml,sle15.yml
Product sle12 has multiple remediations of the same type in rule accounts_tmout: shared.sh,sle12.sh,sle12.yml,sle15.yml
Product sle15 has multiple remediations of the same type in rule accounts_tmout: shared.sh,sle12.sh,sle12.yml,sle15.yml
Product sle12 has multiple remediations of the same type in rule audit_rules_privileged_commands_kmod: sle12.yml,sle15.yml
Product sle15 has multiple remediations of the same type in rule audit_rules_privileged_commands_kmod: sle12.yml,sle15.yml
Traceback (most recent call last):
  File "/home/nico/git/content/./utils/rule_dir_json.py", line 232, in <module>
    main()
  File "/home/nico/git/content/./utils/rule_dir_json.py", line 192, in main
    rule_obj['ovals'], oval_products = handle_ovals(given_products, product_yamls, rule_obj)
  File "/home/nico/git/content/./utils/rule_dir_json.py", line 120, in handle_ovals
    platforms = ssg.oval.applicable_platforms(oval_path)
  File "/home/nico/git/content/ssg/oval.py", line 123, in applicable_platforms
    body = process_file_with_macros(oval_file, subst_dict)
  File "/home/nico/git/content/ssg/jinja.py", line 184, in process_file_with_macros
    return process_file(filepath, substitutions_dict)
  File "/home/nico/git/content/ssg/jinja.py", line 136, in process_file
    return template.render(substitutions_dict)
  File "/home/nico/.pyenv/versions/3.9.6/envs/content_3.9.6/lib/python3.9/site-packages/jinja2/environment.py", line 1304, in render
    self.environment.handle_exception()
  File "/home/nico/.pyenv/versions/3.9.6/envs/content_3.9.6/lib/python3.9/site-packages/jinja2/environment.py", line 925, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "/home/nico/git/content/linux_os/guide/system/software/integrity/software-integrity/aide/aide_check_audit_tools/oval/shared.xml", line 11, in top-level template code
    {{% if 'rhel' not in product %}}
jinja2.exceptions.UndefinedError: 'product' is undefined
```
The prerequisites fail. What do I do in these cases?
For every project I am working on I use a separate pyenv environment.

**Comment by Mab879** [comment]
@truzzon I have fixed a couple more as well.

 What OS are using for this? I'm not reproduce on Fedora 34 with Python 3.9.6.

**Comment by truzzon** [comment]
I am working on Mint 20.2.

I'll keep that tool in mind, when I start to work on sle15 again. I haven't had the time to debug it today.

**Review by Mab879 — APPROVED** [review]

**Comment by Mab879** [comment]
Thank you @truzzon!

**Pull request merged**